### PR TITLE
Add removal of deprecated methods to release notes/upgrading guide

### PR DESCRIPTION
**Pull request by mhajas**

### CODE REVIEW

File: release_notes/topics/20_0_0.adoc [hunk 1]
```diff
@@ -17,3 +17,10 @@ In this release, we are making important changes to `kc.bat` to give the same ex
 
 In this release, the H2 driver has been upgraded from version 1.x to version 2.x.
 See link:{upgradingguide_link}[{upgradingguide_name}] for details on how to migrate.
+
+= Deprecated methods from data providers and models were removed
+
+Several deprecated methods were removed from data providers and models.
+If not done already, their usage needs to be replaced with corresponding
```
Comment (andymunro) [inline release_notes/topics/20_0_0.adoc]:
```suggestion
If not done already, their usage needs to be replaced with the corresponding
```

File: upgrading/topics/keycloak/changes-20_0_0.adoc [hunk 2]
```diff
@@ -27,4 +27,89 @@ See the H2 documentation http://www.h2database.com/html/commands.html#set_non_ke
...
-Purge existing H2 database files to start with an empty database, export and import the realms using Keycloak's export and import functionality, or refer to the http://www.h2database.com/html/migration-to-v2.html[migration notes on the H2 database project's website] for details on how to migrate H2 database contents.
\ No newline at end of file
+Purge existing H2 database files to start with an empty database, export and import the realms using Keycloak's export and import functionality, or refer to the http://www.h2database.com/html/migration-to-v2.html[migration notes on the H2 database project's website] for details on how to migrate H2 database contents.
+
+= Deprecated methods from data providers and models were removed
+
+Prior to Keycloak 15, there was a clean-up of provider and model interfaces
+where we deprecated some methods. Javadoc of these methods contained
+a corresponding replacement method (see Javadoc of Keycloak 19 release).
+In this release the methods were removed. List of all changed classes is below.
```
Comment (andymunro) [inline upgrading/topics/keycloak/changes-20_0_0.adoc]:
```suggestion
In this release the methods were removed. The following is a list of all changed classes.
```

File: upgrading/topics/keycloak/changes-20_0_0.adoc [hunk 3]
```diff
@@ -27,4 +27,89 @@ See the H2 documentation http://www.h2database.com/html/commands.html#set_non_ke
...
+
+Most common patterns for deprecating and now removing the methods are
```
Comment (andymunro) [inline upgrading/topics/keycloak/changes-20_0_0.adoc]:
```suggestion
The most common patterns for deprecating and removing the methods are
```

File: upgrading/topics/keycloak/changes-20_0_0.adoc [hunk 4]
```diff
@@ -27,4 +27,89 @@ See the H2 documentation http://www.h2database.com/html/commands.html#set_non_ke
...
+** `ok.storage.client.ClientLookupProvider`
+** `ok.storage.group.GroupLookupProvider`
+** `ok.storage.user.UserLookupProvider`
+** `ok.storage.user.UserQueryProvider`
+* `server-spi-private` module
+** `ok.events.EventQuery`
+** `ok.events.admin.AdminEventQuery`
+** `ok.keys.KeyProvider`
+
+All changes can be found in the following https://github.com/keycloak/keycloak/pull/14973[pull request].
```
Comment (andymunro) [inline upgrading/topics/keycloak/changes-20_0_0.adoc]: I have not seen a PR in documentation before. Is this valid?

**Review by andymunro — CHANGES_REQUESTED** [review]
Thanks, @mhajas.  A few suggestions.

**Review by hmlnarik — CHANGES_REQUESTED** [review]
Thank you, looks good to me but the link checked does not like the GH line number anchors, could you remove those anchors?

File: upgrading/topics/keycloak/changes-20_0_0.adoc (hunk 4 above)
Comment (mhajas) [reply upgrading/topics/keycloak/changes-20_0_0.adoc]: Adding it there makes sense to me as it contains all the information about what was changed. On the other hand, if it is not common to add it to the upgrading guide I can remove it. @hmlnarik WDYT?

Comment (hmlnarik) [reply upgrading/topics/keycloak/changes-20_0_0.adoc]: Good catch. I agree with @andymunro here, it is common to refer to an issue, should it be needed. Could you please refer to https://github.com/keycloak/keycloak/issues/14720 instead? Something like:
```
All changes are linked to the following https://github.com/keycloak/keycloak/pull/14720[issue].
```

**Review by hmlnarik — DISMISSED** [review]

**Review by hmlnarik — DISMISSED** [review]
Thank you for the PR!

**Review by hmlnarik — APPROVED** [review]

**Pull request merged**